src/app/utils/field-validation/index.ts

```diff
@@ -271,3 +275,223 @@ export const validateField = (
   }
   return ok(true)
 }
+
+/**
+ * Checks if a response is present on a field that is hidden.
+ * The expected behavior is that a response should no be present on a hidden field.
+ * @param response to check for
+ * @param isVisible whether the field is visible
+ * @returns
+ */
+const isResponsePresentOnHiddenFieldV3 = ({
+  formField,
+  response,
+  isVisible,
+  formId,
+}: {
+  formField: FormFieldSchema
+  response: ParsedClearFormFieldResponseV3
+  isVisible: boolean
+  formId: string
+}): Result<boolean, ValidateFieldError> => {
+  if (isVisible) return ok(false)
+
+  if (isGenericStringAnswerResponseV3(response)) {
+    const answer = response.answer
+    const isStringAnswerEmpty = answer.toString().trim() === ''
+    return ok(!isStringAnswerEmpty)
+  }
+  switch (response.fieldType) {
+    case BasicField.YesNo:
+      return ok(response.answer.trim() !== '')
+    case BasicField.Email:
+    case BasicField.Mobile:
+      return ok(
+        response.answer.value.trim() !== '' ||
+          (!!response.answer.signature &&
+            response.answer.signature.trim() !== ''),
+      )
+    case BasicField.Radio:
+      return ok(
+        ('value' in response.answer && response.answer.value.trim() !== '') ||
+          ('othersInput' in response.answer &&
+            response.answer.othersInput.trim() !== ''),
+      )
+    case BasicField.Checkbox:
+      return ok(response.answer.value?.length > 0)
+    case BasicField.Table:
+      return ok(
+        !response.answer.every((rowObject) =>
+          Object.values(rowObject).every((value) => value === ''),
+        ),
+      )
+    case BasicField.Attachment:
+      return ok(
+        response.answer.filename.trim() !== '' || // filename is defined only if there is a file uploaded for the response
+          response.answer.answer.trim() !== '' ||
+          !!response.answer.content,
+      )
+    case BasicField.Children:
+      return ok(
+        (response.answer.child && response.answer.child.length > 1) ||
+          (response.answer.child.length === 1 &&
+            response.answer.child[0] &&
+            response.answer.child[0].every((val) => val !== '')), // If only 1 element which has fields all empty, same as no child selected.
+      )
+  }
+  logInvalidAnswer(formId, formField, 'Invalid response shape')
+  return err(new ValidateFieldError('Response has invalid shape'))
+}
+
+const isValidationRequiredV3 = ({
+  formField,
+  response,
+  isVisible,
+  formId,
+}: {
+  formField: FormFieldSchema
+  response: ParsedClearFormFieldResponseV3
+  isVisible: boolean
+  formId: string
+}): Result<boolean, ValidateFieldError> => {
+  if (isGenericStringAnswerResponseV3(response)) {
+    return ok(
+      (formField.required && isVisible) ||
+        response.answer.toString().trim() !== '',
+    )
+  } else if (response.fieldType === BasicField.YesNo) {
```

Comment:
This is structured this way because that's how `isValidationRequired` is done previously? Looking if we should park this as an item to refactor.

Reply:
refactored to using switch statements